### ISSUE TITLE
Handling click only on part of the trigger component

### ISSUE THREAD

**Issue by nikolayg** [glennflanagan/react-collapsible#9]
Hello,

I have developed my own components for `trigger` and `triggerWhenOpen`. However, the panel collapses/opens whenever I click **anywhere** on the trigger panel. 

I would like open/close only on clicks on specific elements of the trigger area - e.g. a link. I tried to implement this through a custom `handleTriggerClick` function, but couldn't figure out how to detect the source of the event and how to force the component to open/close.

How can I achieve this?

**Comment by cpruijsen** [comment]
@nikolayg I was facing the same issue in an implementation where I had a form inside the tabs, and answering questions would close the collapsible.

I fixed it by having the `open` option on the collapsible take a method that checks for the opened state:

```
<Collapsible toggleOpen={that.toggleOpen} open={that.checkOpen(trigger)} trigger={trigger} key={uuid()}>
```

where toggleOpen is a method which looks for the index of trigger in an array in state (this.state.openTabs) and either pushes the trigger to it, or splices it out, it's invoked on every triggerHandler click after the internal method changes state.

inside Collapsible.js (which I extracted from node_modules and import locally)

```
closeCollapsible: function closeCollapsible() {
    let that = this;
    this.setState({
      isClosed: true,
      shouldSwitchAutoOnNextCycle: true,
      height: this.refs.inner.offsetHeight,
      overflow: 'hidden'
    }, () => {
      that.props.toggleOpen(that.props.trigger);
    });
  },
```

and checkOpen / toggleOpen: 

```
checkOpen(trigger) {
    let index = this.state.openTabs.indexOf(trigger);
    if (index > 0 && this.state.openTabs.length > 0) {
      return true;
    }
    return false;
  }
  toggleOpen(trigger) {
    let openTabs = this.state.openTabs.slice();
    if (openTabs.indexOf(trigger) < 0) {
      openTabs.push(trigger);
    } else {
      let index = openTabs.indexOf(trigger);
      openTabs = openTabs.slice(0, index).concat(openTabs.slice(index + 1));
    }
    this.setState({openTabs: openTabs});
  }
```

Not the cleanest method - but it works ;)

**Comment by glennflanagan** [comment]
Hey guys. I've had a look at this and I'm not sure there is a nice way to do this as I can't figure out a way to get a custom trigger element to fire the handleTriggerClick inside the component. 

It's possible I could restructure the code into an ES6 class which would allow you to extend it.

I'll keep looking into it, and if anyone figures it out feel free to open a PR.

**Comment by awahlfischle** [comment]
I added a prop an Collapsible `classNameToTriggerOpenCloseEvent` and check in `handleTriggerClick` if this class got clicked

Collapsible.js
```
propTypes: {
        ...
        classNameToTriggerOpenCloseEvent: React.PropTypes.string,
       .....
}
....
    handleTriggerClick: function (event) {

        //the user defined his own function, use this
        if (this.props.handleTriggerClick) {
            event.preventDefault();
            this.props.handleTriggerClick(this.props.accordionPosition);
        } else{
            //the user wants the open/close event only for a certain class and provided this
            if(!this.props.classNameToTriggerOpenCloseEvent || this.props.classNameToTriggerOpenCloseEvent === event.target.className){
                this.changeCloseOpenState();
            }
        }

    },

    changeCloseOpenState: function(){
        if (this.state.isClosed === true) {
            this.openCollasible();
        } else {
            this.closeCollapsible();
        }
    },
......
```


CustomComponent.js
```
<Collapsible trigger={trigger} key={props.key} classNameToTriggerOpenCloseEvent={classes.openTriggerClass}>
                {detail}
</Collapsible>
```

**Comment by glennflanagan** [comment]
@awahlfischle That looks interesting. If you'd like, open it up as a PR so I can run the code and if it's ok I'll merge it in.

**Comment by goodmorninggoaway** [comment]
I took a different approach. Here's what I hacked together and works for me. I contemplated contributing back, but it seems a bit hackish. It basically allows you to add whatever arbitraty markup as a sibling to the trigger, but you need to style it so it overlays the trigger.

https://github.com/goodmorninggoaway/react-collapsible/commit/2c3009a42102704d08589e2c994e96dc83801c08

**Comment by glennflanagan** [comment]
Fixed in 1.4.0
